ci-operator/step-registry/rosa/cluster-lease/controller/rosa-cluster-lease-controller-commands.sh

```
@@ -106,12 +106,155 @@ data:
+        rosa list upgrades -c "${cluster_id}" -o json 2>/dev/null \
+            | jq -r --arg desired "${desired_version}" \
+              '.[] | select(.version | startswith($desired)) | .version' 2>/dev/null \
+            | sort -V | tail -n1
+    fi
+}
+
+# Schedule an upgrade for a cluster
+schedule_upgrade() {
+    local cluster_id="$1" type="$2" target_version="$3"
+    if [[ "${type}" == "osd-gcp" ]]; then
+        local schedule_date schedule_time
```

Shellcheck (0.11.0)

[warning] schedule_time appears unused. Verify use (or export if used externally).

(SC2034)